### PR TITLE
Ensure that test titles are strings before sanitizing them for screenshot names

### LINKED ISSUE #4310: 'Cannot read property \'replace\' of null' from sanitize-filename err being thrown
We are using Docker containers to run multiple test cases in parallel. Each container run a single test. we are launching multiple test cases together to test the same application located in another server. When I launch a single test it passes. Problems start when I run the same test 4 or 5 tests in parallel. Some of them pass the others fail.

The test is simple it enters the login and password in the login webpage and clicks "connect" and wait for the home page.

I read the access log of the server and I found that the number of requests is not the same for each client ! I found that weird because I waited for the same behaviour from the web server.

I extracted the cypress Debug logs and I found that the failed clients have this error:

```
cypress:server:reporter got mocha event 'fail' with args: [ 
  { 
    id: 'r3', 
    err: { 
      message: 'Cannot read property \'replace\' of null', 
      name: 'TypeError', 
      stack: 'TypeError: Cannot read property \'replace\' of null\n    
        at sanitize (/root/.cache/Cypress/3.3.1/Cypress/resources/app/packages/server/node_modules/sanitize-filename/index.js:41:6)\n    
        at module.exports (/root/.cache/Cypress/3.3.1/Cypress/resources/app/packages/server/node_modules/sanitize-filename/index.js:51:16)\n
```

I tried to run those tests not in containers they passed. 

I couldn't understand the source of the problem, eather the cypress test or the application itself.

Any help will be appreciated.

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- Close #4310 
- Write tests for many non-string values to ensure screenshot is taken.

### CODE REVIEW

File: packages/server/lib/screenshots.coffee [hunk 1]
```diff
@@ -309,7 +309,10 @@ getPath = (data, ext, screenshotsFolder) ->
   if data.name
     names = data.name.split(pathSeparatorRe).map(sanitize)
   else
-    names = [data.titles.map(sanitize).join(RUNNABLE_SEPARATOR)]
+    ## they could have test titles that are not strings, 
+    ## remove those before trying to sanitize
+    stringTitles = _.filter(data.titles, _.isString)
```
Comment (chrisbreiding) [inline packages/server/lib/screenshots.coffee]: Maybe we should stringify the value instead of removing it? Any thoughts @brian-mann?

Comment (jennifer-shehane) [reply packages/server/lib/screenshots.coffee]: Yeah, this is what @brian-mann said also. 😅 I will update.

**Comment by Bkucera** [comment]
@jennifer-shehane why are we allowing non-string test titles anyways? Then we wouldn't need to add this logic or ever account for non-string test names again

**Comment by jennifer-shehane** [comment]
@Bkucera They already work. They work fine when running tests in Cypress and other frameworks, it's just when we try to translate the title to a screenshot filename that it borks.

**Comment by Bkucera** [comment]
@jennifer-shehane how does it look in the dashboard though? reporters? I don't think we handle non-strings anywhere else really. If there's a discussion about this somewhere i'd like to read it

**Comment by jennifer-shehane** [comment]
We're converting them all to strings, so it wouldn't display any other way in the Dashboard. I have no idea how reporters handle it.

**Review by brian-mann — APPROVED** [review]

**Comment by brian-mann** [comment]
@Bkucera 	regarding test titles which are purely `null` or `undefined` that's a separate discussion.

`undefined` will not work in JSON, whereas `null` will. more than likely `mocha` shits the bed if it's a mess, so we may not handle it because it may not be possible.

**Pull request merged**